### ISSUE TITLE
The weather is not randomized following the input config of the init function

### ISSUE THREAD

**Issue by pengzhenghao** [opendilab/DI-drive#9]
## ISSUE TEMPLATE

- [ ] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [ ] system worker bug
    + [ ] system utils bug
    + [ ] code design/refactor
    + [ ] documentation request
    + [x] new feature request
- [ ] I have visited the [readme](https://github.com/opendilab/DI-drive/blob/main/README.md) and [doc](https://opendilab.github.io/DI-drive/)
- [ ] I have searched through the [issue tracker](https://github.com/opendilab/DI-drive/issues) and [pr tracker](https://github.com/opendilab/DI-drive/pulls)
- [ ] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import core, torch, sys
  print(core.__version__, torch.__version__, sys.version, sys.platform)
  ```

In the following code in carla_simulator.py, it seems that the weather is not randomized following the input `config` of the `init` function.

![Screenshot from 2021-12-07 21-37-30](https://user-images.githubusercontent.com/22206995/145039124-74574063-91d0-4e75-a196-fce2ab2ec51f.png)

You can see that the config specifies a subset of the whethers.

![Screenshot from 2021-12-07 21-39-27](https://user-images.githubusercontent.com/22206995/145039401-165fe4af-ed35-4c72-8821-08ae60f76182.png)

**Comment by pengzhenghao** [comment]
It is also possible that I misunderstand the "weathers" in config. Because I also find "whether" in the config.

**Comment by RobinC94** [comment]
If you are using benchmark setting (i.e. FullTown01-v1 or NoCrash), the default weather config in simulator will be blocked. So the actual weathers will be decided by benchmark settings.
We also provide a 'weathers' key in benchmark collector\evaluator which is used to change the default weather setting in benchmark. For example a user wants to run FullTown01's routes with a provided list of weathers not defined in existed benchmark settings. It can be manually set by change the 'weathers' key in benchmark collector\evaluator.
So in short the priority of weather configs should be: benchmark config ('weathers' in benchmark) > benchmark default > simulator config ('weather' in simulator)
